src/boot/efi/smbios.c

```diff
@@ -181,3 +201,49 @@ const char* smbios_find_oem_string(const char *name) {
 
         return NULL;
 }
+
+static const char* smbios_get_string(const SmbiosHeader *header, size_t nr, size_t left) {
+        assert(header);
+        const char *s = (const char *) header;
+
+        assert(left >= header->length);
+        s += header->length;
+        left -= header->length;
+
+        size_t index = 1;
+        for (const char *p = s; index <= nr && p < s + left; index++) {
+                const char *e = memchr(p, 0, s + left - p);
+
```

Comment:
please keep fnction calls and their immediate return value/error handling together, i.e. drop the empty line before this if check.

as per coding style